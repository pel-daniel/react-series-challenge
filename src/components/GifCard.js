import React from 'react';

import '../styles/GifCard.css';

const GifCard = (props) => {
  var { images, title, user } = props;

  return (
    <div className="GifCard">
      { user &&
        <a href={ user.profile_url } className="GifCardUser">
          <img src={ user.avatar_url } alt={ user.username } />
        </a>
      }
      <img src={ images.original.url } alt={title} />
    </div>
  );
};

export default GifCard;
