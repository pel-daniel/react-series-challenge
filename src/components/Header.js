import React from 'react';
import { NavLink } from 'react-router-dom';

import {
  HeaderWrapper,
  HeaderContainer,
  HeaderLogo,
  HeaderNavItems
} from '../styles/Header.style.js';

import logo from '../icons/logo.svg';

const Header = () => {
  return (
    <HeaderWrapper>
      <div className="App-container">
        <HeaderContainer>
          <NavLink to="/">
            <HeaderLogo src={logo} alt="logo" />
          </NavLink>

          <HeaderNavItems>
            <NavLink to="/favorites">Favorites</NavLink>
          </HeaderNavItems>
        </HeaderContainer>
      </div>
    </HeaderWrapper>
  )
}

export default Header;
