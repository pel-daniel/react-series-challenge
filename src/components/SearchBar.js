import React from 'react';
import { connect } from 'react-redux'

import '../styles/SearchBar.css';
import gifActions from '../actions/gifActions';

const SearchBar = ({ dispatch }) => {
  let input;

  return (
    <div className="SearchBar">
      <input
        type="text"
        placeholder="Search gifs!!"
        ref={ node => input = node }
        onKeyPress={ event => {
          if(event.key === 'Enter' && input.value.trim()) {
            dispatch(gifActions.creators.searchGifs(input.value));
          }
        }}
      />
    </div>
  )
}

export default connect()(SearchBar);
