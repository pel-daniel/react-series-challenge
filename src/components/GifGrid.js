import React from 'react';
import { connect } from 'react-redux';

import GifCard from './GifCard';

import '../styles/GifGrid.css';

const GifGrid = (props) => {
  const { gifs } = props;

  return (
    <div className="GifGrid mt">
      { gifs.map((gif) =>
        <GifCard key={gif.id} {...gif} />
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  const { gifs } = state;

  return {
    gifs
  };
};

const mapDispatchToProps = (dispatch) => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(GifGrid);
