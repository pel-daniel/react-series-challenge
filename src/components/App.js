import React, { Component } from 'react';
import { Provider } from 'react-redux';

import store from '../store';

import Header from '../components/Header';
import SearchBar from '../components/SearchBar';
import '../styles/App.css';

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <Header />

          <div className="App-container mt">
            <SearchBar />
          </div>
        </div>
      </Provider>
    );
  }
}

export default App;
