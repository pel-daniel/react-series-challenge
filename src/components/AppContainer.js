import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom';

import gifActions from '../actions/gifActions';
import GifGrid from '../components/GifGrid';
import SearchBar from '../components/SearchBar';

class AppContainer extends Component {
  componentDidMount() {
    const { dispatch } = this.props;

    dispatch(gifActions.creators.getTrendingGifs());
  }

  render() {
  const { gifs, favorites } = this.props;
  const favoriteGifs = gifs.filter((gif) => favorites.indexOf(gif.id) !== -1);

  return (
    <div className="AppContainer mt">
      <SearchBar />

      <Route
        exact
        path="/"
        render={({ match }) => (
          <GifGrid gifs={ gifs } />
        )}
      />
      <Route
        path="/favorites"
        render={({ match }) => (
          <GifGrid gifs={ favoriteGifs } />
        )}
      />
    </div>
  );
  }
}

const mapStateToProps = (state) => state;

export default connect(mapStateToProps)(AppContainer);
