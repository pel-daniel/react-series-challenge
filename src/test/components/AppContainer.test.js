import React from 'react';
import { shallow } from 'enzyme';

import { AppContainer } from '../../components/AppContainer';

describe('Header component', () => {
  test('should shallow correctly', () => {
    const getTrendingGifs = () => {};
    const header = shallow(<AppContainer getTrendingGifs={ getTrendingGifs } />);

    expect(header).toMatchSnapshot();
  });
});
