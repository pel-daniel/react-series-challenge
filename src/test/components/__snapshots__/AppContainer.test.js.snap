// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should shallow correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppContainer
    getTrendingGifs={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(SearchBar) />,
        <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/favorites"
          />
</Switch>,
      ],
      "className": "mt",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/favorites"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(SearchBar) />,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        path="/favorites"
            />
</Switch>,
        ],
        "className": "mt",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                path="/favorites"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
